src/cryptoadvance/specter/controller.py

```diff
@@ -467,21 +468,122 @@ def new_wallet(wallet_type):
 
     if request.method == 'POST':
         action = request.form['action']
-        wallet_name = request.form['wallet_name']
-        if wallet_name in app.specter.wallet_manager.wallets_names:
-            err = "Wallet already exists"
-        address_type = request.form['type']
-        pur = {
-            '': "General",
-            "wpkh": "Segwit (bech32)",
-            "sh-wpkh": "Nested Segwit",
-            "pkh": "Legacy",
-            "wsh": "Segwit (bech32)",
-            "sh-wsh": "Nested Segwit",
-            "sh": "Legacy",
-        }
-        sigs_total = int(request.form.get('sigs_total', 1))
-        sigs_required = int(request.form.get('sigs_required', 1))
+        if action == "importwallet":
+            wallet_data = json.loads(request.form['wallet_data'])
+            wallet_name = wallet_data['label']
+            startblock = wallet_data['blockheight']
```

Comment:
If this information is not in json - let's set defaults - for example "Imported Wallet" for wallet_name and current blockheight for startblock (so we don't trigger rescan)